### PR TITLE
Update factions.go

### PR DESCRIPTION
**Pull request by adrianvrj**
- [x] issue #217 
- [x] follows contribution [guide](https://github.com/keep-starknet-strange/art-peace/blob/main/CONTRIBUTING.md)
- [ ] code change includes tests
- [ ] breaking change

### CODE REVIEW

File: backend/routes/factions.go [hunk 1]
```diff
@@ -202,3 +204,68 @@ func getFactions(w http.ResponseWriter, r *http.Request) {
...
+
+	query := `
+		SELECT user_address, allocation
+		FROM FactionMembersInfo
+		WHERE faction_id = $1
+		ORDER BY allocation DESC
+		LIMIT $2 OFFSET $3
+	`
+
+	rows, err := core.ArtPeaceBackend.Databases.Postgres.Query(context.Background(), query, factionID, pageLength, offset)
```
Comment (b-j-roberts) [inline backend/routes/factions.go:237]: Lines 237-268 can be cleaned up, if you use the `PostgresQueryJson` with a custom faction members type, like this example: https://github.com/keep-starknet-strange/art-peace/blob/5f5aa850349a1e83ac1d632ea8d4a3e6e80f00fd/backend/routes/factions.go#L161

File: backend/routes/factions.go [hunk 2]
```diff
@@ -202,3 +204,68 @@ func getFactions(w http.ResponseWriter, r *http.Request) {
...
+		pageLength = 50
+	}
+
+	page, err := strconv.Atoi(r.URL.Query().Get("page"))
+	if err != nil || page <= 0 {
+		page = 1
+	}
+	offset := (page - 1) * pageLength
+
+	query := `
```
Comment (b-j-roberts) [inline backend/routes/factions.go:229]: For this query, can you add a JOIN statement with the `Users` table to get the user's username aswell as part of the return value.


Also, it is possible for multiple rows in `FactionMembersInfo` to have the same user address. Can you have it so the allocation is accumulated ( add up all allocations for all rows with the same user address )

So for example:
user address | allocation
0x12 | 3
0x24 | 4
0x12 | 2
0x35 | 1

Would give
user address | allocation
0x12 | 5
0x24 | 4
0x35 | 1

**Review by b-j-roberts — COMMENTED** [review]
Nice work so far!

**Comment by adrianvrj** [comment]
> Nice work so far!

will solve your requests asap!

**Comment by b-j-roberts** [comment]
When using the route, I am getting `{"data": null}.` So I think there may be something wrong with the query. 

You can test the route by starting the app with `docker compose`. There are details in the readme for that, then doing
`curl http://localhost:8080/get-faction-members\?factionID\=0`

I'd also recommend testing the query with `psql` so you don't need to restart each time you change the query.

**Comment by adrianvrj** [comment]
> When using the route, I am getting `{"data": null}.` So I think there may be something wrong with the query.
> 
> You can test the route by starting the app with `docker compose`. There are details in the readme for that, then doing `curl http://localhost:8080/get-faction-members\?factionID\=0`
> 
> I'd also recommend testing the query with `psql` so you don't need to restart each time you change the query.

That might happen cause users table is empty
image
Once I added some users manually to the table this is what I get.

**Comment by b-j-roberts** [comment]
I see, thanks.
So the users table won't always have usernames saved for all addresses, so it should be be setup to return an empty string as the username when that's the case.
I think using COALESCE for the username should work, but might need to test it out

**Comment by adrianvrj** [comment]
> I see, thanks. So the users table won't always have usernames saved for all addresses, so it should be be setup to return an empty string as the username when that's the case. I think using COALESCE for the username should work, but might need to test it out

Will test it out and lyk

**Comment by adrianvrj** [comment]
looking like this now without any users in the table!
image
@b-j-roberts

**Comment by b-j-roberts** [comment]
Great work, I appreciate the images showing your tests output

**Pull request merged**